src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive.splice(Animal.alive.indexOf(this), 1);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(victim) {
+    if (victim instanceof Carnivore || victim.hidden) {
+      return;
+    }
+    victim.decreaseHealth(50);
```

Comment:
I'd advise to move details of reaction to bite inside of `Herbivore` class. So that if you want to change herbivores' reaction to bite, you go and make this change inside `Herbivore` class, not `Carnivore` one. Health reduction by 50 is herbivores feature, so let's keep it in their class. It could be something like this:
```
class Herbivore extends Animal {
  ...
  receiveBite() {
    if (!this.hidden) {
      this.decreaseHealth(HERBIVORE_HEALTH_DECREASE); // also define 50 as constant
    }
  }
}
...
class Carnivore extends Animal {
  ...
  bite(victim) {
    if (victim instanceof Carnivore) {
      return;
    }
    victim.receiveBite();
  }
}
```